### DISCUSSION TITLE
如何add_requires添加的package里的静态库合并到最终静态库

### DISCUSSION THREAD

**Discussion by xixioba** [xmake-io/xmake#5221] [Q&A] [answered]
build.merge_archive似乎只能合并add_deps，但是没法合并add_packages的库，有什么办法么，比如合并boost静态库

**Comment by waruqi** [comment]
自己调用 merge_staticlibs 写 rules 取 package libs 合并

https://github.com/xmake-io/xmake/blob/master/xmake/rules/utils/merge_archive/xmake.lua

**Comment by xixioba** [comment]
静态库合并到动态库里面呢😅

**Reply by waruqi** [reply]
动态库还要合并来干嘛，直接走 link 不就完事了么。。

**Reply by xixioba** [reply]
的确这么走是最简单的，我的案例是
1. 生成了一个静态库 lib_a.a
2. 生成一个动态库(lib_a.so)，依赖这个静态库
我希望是直接生成lib_a.so，而不是再去定义一圈依赖项

**Reply by waruqi** [reply, marked as answer]
建议不要瞎折腾，规规矩矩来

**Comment by xixioba** [comment]
另外，生成的静态库debug信息正常，动态库似乎缺少了debug信息，这是什么原因呢

**Reply by waruqi** [reply]
要看你是 debug 还是 release 编译了，对应 rules配置了没有，然后检查下 link flags 看是否 strip symbols了

**Reply by xixioba** [reply]
很奇怪的是，全局debug模式，.a库可以带debug信息，但是.so就没，我是macos系统，.so target里strip none也没有效果

**Reply by xixioba** [reply]
大无语，似乎是strip不小心给干掉了
